packages/i18nify-js/src/modules/phoneNumber/__tests__/getMaskedPhoneNumber.test.ts

```diff
@@ -0,0 +1,152 @@
+    const options = {
+      countryCode: 'US',
+      phoneNumber: '2345678901',
+      withDialCode: false,
+      maskingOptions: {
+        maskingStyle: MaskingStyle.Prefix,
+        maskedDigitsCount: 6,
+      },
+    };
+    const expected = 'xxx-xxx-8901';
+    expect(
+      getMaskedPhoneNumber(options as GetMaskedPhoneNumberOptions),
+    ).toEqual(expected);
+  });
+
+  it('should return phone number masked with suffix', () => {
+    const options = {
+      countryCode: 'US',
+      phoneNumber: '2345678901',
+      withDialCode: false,
+      maskingOptions: {
+        maskingStyle: MaskingStyle.Suffix,
+        maskedDigitsCount: 6,
+      },
+    };
+    const expected = '234-5xx-xxxx';
+    expect(
+      getMaskedPhoneNumber(options as GetMaskedPhoneNumberOptions),
+    ).toEqual(expected);
+  });
+
+  it('should handle alternate masking of digits', () => {
+    const options = {
+      countryCode: 'US',
+      phoneNumber: '2345678901',
+      withDialCode: false,
+      maskingOptions: {
+        maskingStyle: MaskingStyle.Alternate,
+        maskingChar: '*',
+      },
+    };
+    const expected = '2*4*6*8*0*';
+    expect(
+      getMaskedPhoneNumber(options as GetMaskedPhoneNumberOptions),
+    ).toEqual(expected);
+  });
+
+  it('should handle masking with just countryCode', () => {
+    const options = {
+      countryCode: 'US',
+      withDialCode: true,
+    };
+    const expected = '+1 xxx-xxx-xxxx';
+    expect(
+      getMaskedPhoneNumber(options as GetMaskedPhoneNumberOptions),
+    ).toEqual(expected);
+  });
+
+  it('should return phone number with full masking and dial code when maskingChar is #', () => {
+    const options = {
+      countryCode: 'US',
+      phoneNumber: '2345678901',
+      withDialCode: true,
+      maskingOptions: { maskingStyle: MaskingStyle.Full, maskingChar: '#' },
+    };
+    const expected = '+1 ###-###-####';
+    expect(
+      getMaskedPhoneNumber(options as GetMaskedPhoneNumberOptions),
+    ).toEqual(expected);
+  });
+
+  it('should return formatted phone number with complete masking when no masking options provided', () => {
+    const options = {
+      countryCode: 'US',
+      phoneNumber: '2345678901',
+      withDialCode: true,
+    };
+    const expected = '+1 xxx-xxx-xxxx';
+    expect(
+      getMaskedPhoneNumber(options as GetMaskedPhoneNumberOptions),
+    ).toEqual(expected);
+  });
+
+  it('should handle input with non-numeric characters in phoneNumber', () => {
+    const options = {
+      countryCode: 'US',
+      phoneNumber: '+1 (234) 567-8901',
+      withDialCode: false,
+      maskingOptions: { maskingStyle: MaskingStyle.Full, maskingChar: '*' },
+    };
+    const expected = '***-***-****';
+    expect(
+      getMaskedPhoneNumber(options as GetMaskedPhoneNumberOptions),
+    ).toEqual(expected);
+  });
+
+  it('should perform complete masking if maskedDigitsCount is larger than phoneNumber length', () => {
```

Comment:
Can you write 2 more test cases
1. should mask with just phone number (without dialcode) without countryCode. 
2. should mask with just phone number (with dialcode) without countryCode